### PR TITLE
Fix of Issue #276

### PR DESCRIPTION
**Pull request by dongbohu**
This PR fixes issue #276. It adds paddings of `0`s into some signature names so that when they are shown on the web, the alphabetical order is consistent to the numerical order. Three files are revised:
(1) data/sample_signature_activity.txt: All signature names (on the first row) are padded.
(2) data/signature_gene_network.txt: All signature names (on the first column) are padded.
(3) data/README.md: added some description text on the padding for users who will load their own data files. I also tweaked another line in this file to make it more readable (hopefully). I hope @rzelayafavila won't mind.

### CODE REVIEW

File: data/README.md [hunk 1]
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
   Contains signature (node) activity levels (as absolute values) for each
   sample in the compendium derived from the eADAGE machine learning model.
   In this file, samples are rows, and signatures are columns. This file is
-  based on the
-  `all-pseudomonas-gene-normalized_HWActivity_perGene_with_net300_100models_1_100_k=300_seed=123_ClusterByweighted_avgweight_network_ADAGE.txt`
-  file, which is now in the `data/old/` folder and was generated by
+  based on:
+  ```
+  all-pseudomonas-gene-normalized_HWActivity_perGene_with_net300_100models_1_100_k=300_seed=123_ClusterByweighted_avgweight_network_ADAGE.txt
```
Comment (dongbohu) [inline data/README.md]: the main reason why I changed this line was because originally the super-long file name and description text were mingled together and seemed a little messy.

File: data/README.md [hunk 2]
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
...
+  `Sig9` into `Sig009`).
+
 ### **gene_gene_network_cutoff_0.2.txt.gz**
   Contains gene-gene network information derived from the eADAGE machine
   learning model, where each row contains the edge weight between two genes
   in this model and whether this is a positive or negative weight. Because
   of its size, this file has been compressed using
   [gzip](http://www.gzip.org/). However, its decompressed content is a
-  tab-delimited text file as well. 
+  tab-delimited text file as well.
```
Comment (dongbohu) [inline data/README.md]: This trailing space was corrected automatically by emacs.

**Comment by dongbohu** [comment]
Here is a demo page that includes the new signature names:
http://165.123.67.202/#/signature/search?mlmodel=7

File: data/README.md (hunk 1 above)
Comment (rzelayafavila) [reply data/README.md]: Sure, I agree that this looks better!

File: data/README.md [hunk 3]
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
...
-  file, which is now in the `data/old/` folder and was generated by
+  based on:
+  ```
+  all-pseudomonas-gene-normalized_HWActivity_perGene_with_net300_100models_1_100_k=300_seed=123_ClusterByweighted_avgweight_network_ADAGE.txt
+  ```
+
+  which is now in the `data/old/` folder and was generated by
   [this script](https://bitbucket.org/greenelab/eadage/src/tip/node_interpretation/write_HWactivity.R).
 
+  **Note:** The signatures shown on the web (click
```
Comment (rzelayafavila) [inline data/README.md]: Great docs! Do you think it would be better to say "... on the web interface" or "... on the web UI" to be more specific?

File: data/README.md [hunk 4]
```diff
@@ -67,6 +76,10 @@ instance, they should be in the same format as the files below.
   where each row represents a gene set, except this file does not include a
   description column, which is the second column in the GMT file format.
 
+  As we mentioned in the description of *sample_signature_activity.txt*,
+  depending on your preference of the signatures' order shown online, you may
+  need to pad some signatures' names.
```
Comment (rzelayafavila) [inline data/README.md]: Hmmm, I think this sentence is a little hard to follow. Maybe just changing the order and words a bit to something like "As we mentioned in the description of *sample_signature_activity.txt*, you may
need to pad some signatures' names depending on your preferred order of the signatures in the web interface."

File: data/sample_signature_activity.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
...	Sig181neg	Sig182neg	Sig183neg	Sig184neg	Sig185neg	Sig186neg	Sig187neg	Sig188neg	Sig189neg	Sig190neg	Sig191neg	Sig192neg	Sig193neg	Sig194neg	Sig195neg	Sig196neg	Sig197neg	Sig198neg	Sig199neg	Sig200neg	Sig201neg	Sig202neg	Sig203neg	Sig204neg	Sig205neg	Sig206neg	Sig207neg	Sig208neg	Sig209neg	Sig210neg	Sig211neg	Sig212neg	Sig213neg	Sig214neg	Sig215neg	Sig216neg	Sig217neg	Sig218neg	Sig219neg	Sig220neg	Sig221neg	Sig222neg	Sig223neg	Sig224neg	Sig225neg	Sig226neg	Sig227neg	Sig228neg	Sig229neg	Sig230neg	Sig231neg	Sig232neg	Sig233neg	Sig234neg	Sig235neg	Sig236neg	Sig237neg	Sig238neg	Sig239neg	Sig240neg	Sig241neg	Sig242neg	Sig243neg	Sig244neg	Sig245neg	Sig246neg	Sig247neg	Sig248neg	Sig249neg	Sig250neg	Sig251neg	Sig252neg	Sig253neg	Sig254neg	Sig255neg	Sig256neg	Sig257neg	Sig258neg	Sig259neg	Sig260neg	Sig261neg	Sig262neg	Sig263neg	Sig264neg	Sig265neg	Sig266neg	Sig267neg	Sig268neg	Sig269neg	Sig270neg	Sig271neg	Sig272neg	Sig273neg	Sig274neg	Sig275neg	Sig276neg	Sig277neg	Sig278neg	Sig279neg	Sig280neg	Sig281neg	Sig282neg	Sig283neg	Sig284neg	Sig285neg	Sig286neg	Sig287neg	Sig288neg	Sig289neg	Sig290neg	Sig291neg	Sig292neg	Sig293neg	Sig294neg	Sig295neg	Sig296neg	Sig297neg	Sig298neg	Sig299neg	Sig300neg
```
Comment (rzelayafavila) [inline data/sample_signature_activity.txt]: So were these padding zeroes added manually (with a Python script or something else)?

File: data/signature_gene_network.txt [hunk 6]
```diff
@@ -1,102 +1,102 @@
-Sig1pos	PA0039	PA0139	PA0376	PA0409	PA0456	PA0576	PA0579	PA0762	PA0805	PA0865	PA0973	PA0996	PA1000	PA1159	PA1288	PA1544	PA1579	PA1582	PA1583	PA1588	PA1592	PA1777	PA1800	PA1801	PA1802	PA1804	PA1830	PA1847	PA2623	PA2634	PA2741	PA2742	PA2743	PA2853	PA2966	PA3031	PA3049	PA3162	PA3201	PA3202	PA3244	PA3385	PA3529	PA3533	PA3637	PA3655	PA3656	PA3745	PA3814	PA3815	PA3819	PA4067	PA4239	PA4241	PA4242	PA4243	PA4244	PA4246	PA4247	PA4248	PA4249	PA4250	PA4251	PA4253	PA4254	PA4257	PA4258	PA4259	PA4262	PA4263	PA4264	PA4265	PA4266	PA4267	PA4268	PA4272	PA4273	PA4315	PA4366	PA4433	PA4463	PA4464	PA4563	PA4568	PA4578	PA4661	PA4671	PA4741	PA4745	PA4761	PA4922	PA4942	PA4944	PA5013	PA5148	PA5178	PA5240	PA5257	PA5276	PA5300	PA5301	PA5316	PA5337	PA5348	PA5438	PA5461	PA5491	PA5555	PA5558	PA5559	PA5560	PA5569	PA5570
```
Comment (rzelayafavila) [inline data/signature_gene_network.txt]: Also here, how was this file modified?

File: data/README.md [hunk 7]
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
...
+  based on:
+  ```
+  all-pseudomonas-gene-normalized_HWActivity_perGene_with_net300_100models_1_100_k=300_seed=123_ClusterByweighted_avgweight_network_ADAGE.txt
+  ```
+
+  which is now in the `data/old/` folder and was generated by
   [this script](https://bitbucket.org/greenelab/eadage/src/tip/node_interpretation/write_HWactivity.R).
 
+  **Note:** The signatures shown on the web (click
+  [here](http://adage.greenelab.com/#/signature/search?mlmodel=1) to see a
```
Comment (mhuyck) [inline data/README.md]: Minor quibble: could this say

```markdown
...(view [this page](http://adage.greenelab.com/#/signature/search?mlmodel=1) to
see an example)...
```

Someone sensitized me to that fact that not everyone uses a mouse anymore and it looks funny to see that written when you actually need to tap or use some assistive technology to open the link another way. Also, I think it just sounds better.

File: data/README.md [hunk 8]
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
...
+  demo) will be in **alphabetical order**. If you would like them to be ranked
```
Comment (mhuyck) [inline data/README.md]: Saying "ranked" is technically correct, but I think "sorted" is a more common way to put this and would sound more natural.

File: data/README.md [hunk 9]
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
...
+  in another order, you may have to pad some signatures' name (such as changing
```
Comment (mhuyck) [inline data/README.md]:
```
...you may have to pad the numbers in the names for some signatures...
```

File: data/README.md (hunk 4 above)
Comment (mhuyck) [inline data/README.md]: Slightly easier to read if we cut out some extra words and order the phrases this way:

```
As mentioned in *sample_signature_activity.txt* above, you may wish to pad numbers
in signature names so they will be sorted in a useful way.
```

File: data/README.md (hunk 3 above)
Comment (dongbohu) [reply data/README.md]: Sure. I will change "web" into "web interface".

File: data/README.md (hunk 4 above)
Comment (dongbohu) [reply data/README.md]: @rzelayafavila: Yes, I agree this sentence reads a little awkward. @mhuyck suggested another alternative which I think is great. If you also agree, I will change to his version.

File: data/sample_signature_activity.txt (hunk 5 above)
Comment (dongbohu) [reply data/sample_signature_activity.txt]: Yes, I wrote a 4-line python script that generated the new row, then replaced the original row with it. Same idea with the other data file, but I used `cut/paste` command to replace the column.

File: data/signature_gene_network.txt (hunk 6 above)
Comment (dongbohu) [reply data/signature_gene_network.txt]: I used `cut` command to copy the file w/o the first command, then created a new column with a simple Python script, then used `paste` command to clue these two parts together.

File: data/README.md (hunk 7 above)
Comment (dongbohu) [reply data/README.md]: Good point. Sure!

File: data/README.md (hunk 8 above)
Comment (dongbohu) [reply data/README.md]: Hmm, originally I thought `sorted` sounds a little too "techie". I will change it to "sorted" then.

File: data/README.md (hunk 4 above)
Comment (dongbohu) [reply data/README.md]: How about `... in a more useful way`?

File: data/README.md (hunk 8 above)
Comment (mhuyck) [reply data/README.md]: Hah! I had the opposite impression. Either way is fine with me... it's a minor thing.

File: data/README.md (hunk 4 above)
Comment (mhuyck) [reply data/README.md]: My English teacher would have pressed me on this: more useful than *what*? If we aren't going into detail on that, I think it makes sense to caution our readers to ensure that an alphabetical sort will produce something useful.

File: data/signature_gene_network.txt (hunk 6 above)
Comment (dongbohu) [reply data/signature_gene_network.txt]: The procedure is something like this:
```
cut -f2- signature_gene_network.txt > data_wo_col1.txt
# use python to generate a new column with filename "new_col1.txt" ...
paste new_col1.txt data_wo_col1.txt > signature_gene_network.txt
```
In `cut/paste` commands, the default delimiter is tab. (You can customize it of course.)

File: data/README.md (hunk 8 above)
Comment (dongbohu) [reply data/README.md]: Instead of "sorted" or "ranked", maybe we should use the more generic word "displayed".

Comment (rzelayafavila) [reply data/README.md]: "If you would like them ordered differently..." is another option.

File: data/README.md (hunk 4 above)
Comment (rzelayafavila) [reply data/README.md]: Sure, sounds good

Comment (rzelayafavila) [reply data/README.md]: Should we include the fact that we have added this padding to these two files available in this repository (in this section and the section for *sample_signature_activity.txt* above)?

File: data/signature_gene_network.txt (hunk 6 above)
Comment (rzelayafavila) [reply data/signature_gene_network.txt]: Ok, gotcha. I was wondering if we should tell the users how we made these modifications for these files, but I guess that is up to them, as long as they know that we have modified these files.

File: data/README.md (hunk 4 above)
Comment (dongbohu) [reply data/README.md]: Probably not, for two reasons:
(1) These new instructions are for users who want to load their own data, not users who will be using our data (such as Hogan Lab);
(2) The demo page already shows that we have padded our signatures.

**Comment by dongbohu** [comment]
@rzelayafavila and @mhuyck: I think I have addressed all your comments. Please take a look again and let me know if you have any other concerns. Thanks.
